src/frontend/src/flows/manage/index.ts

```diff
@@ -271,6 +282,20 @@ export const renderManage = async ({
 
 export const displayManagePage = renderPage(displayManageTemplate);
 
+function isPinAuthenticated(
+  devices_: DeviceData[],
+  connection: AuthenticatedConnection
+): boolean {
+  const principal = connection.identity.getPrincipal();
+  const currentDevice = devices_.find((device) => {
+    const principal1 = Principal.selfAuthenticating(
```

Comment:
this can throw an exception no?

Reply:
I don't think so... It calculates the sha of the provided `Uint8Array` and returns it as a principal. Any `Uint8Array` is accepted.